tests/std/include/test_vector_algorithms_support.hpp

```diff
@@ -41,7 +42,32 @@ inline void initialize_randomness(std::mt19937_64& gen) {
 extern "C" long __isa_enabled;
 
 inline void disable_instructions(ISA_AVAILABILITY isa) {
-    __isa_enabled &= ~(1UL << static_cast<unsigned long>(isa));
+    const unsigned long as_ulong = static_cast<unsigned long>(isa);
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#else // ^^^ defined(__clang__) / !defined(__clang__) vvv
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#endif // ^^^ !defined(__clang__) ^^^
+    const char* const env_val = std::getenv("STL_TEST_DOWNLEVEL_HOST");
+#ifdef __clang__
+#pragma clang diagnostic pop
+#else // ^^^ defined(__clang__) / !defined(__clang__) vvv
+#pragma warning(pop)
+#endif // ^^^ !defined(__clang__) ^^^
+
+    if (env_val == nullptr || std::atoi(env_val) == 0) {
+        if ((__isa_enabled & (1UL << as_ulong)) == 0) {
+            std::printf("The feature %lu is not available, the test does not have full coverage!\n", as_ulong);
+            std::printf("You can set environment variable STL_TEST_DOWNLEVEL_HOST to 1,\n"

```

Comment:
These could really be a single printf. It's strange to use two printfs for three lines.